### ISSUE TITLE
[1.6.0rc1] Print Time Left Shows Wrong value

### ISSUE THREAD

**Issue by mrspaceman** [OctoPrint/OctoPrint#4070]
### The problem

I start a print and the "Approx Total Print Time" shows a value similar to my slicer (Cura 4.8.0)

part way through a print (exactly when varies) the "Print Time Left" will increase to a ridiculous value

### Did the issue persist even in safe mode?

Yes, it did persist

### Version of OctoPrint

OctoPrint: 1.6.0rc1 

### Operating system running OctoPrint

Rasbian

### Printer model & used firmware incl. version

Ender 3 v2

### Browser and version of browser, operating system running browser

chrome Version 89.0.4389.90 (Official Build) (64-bit)

### Checklist of files to include below

- [X] octoprint.log
- [X] serial.log or contents of your terminal tab (always include in cases of issues with printer communication)
- [X] Contents of the JavaScript browser console (always include in cases of issues with the user interface)
- [X] Screenshots and/or videos showing the problem (always include in case of issues with the user interface)
- [X] GCODE file with which to reproduce (always include in case of issues with GCODE analysis or printing behaviour)

[Screenshot showing times left](https://user-images.githubusercontent.com/378813/113262333-65904400-92c8-11eb-87bd-17b004ce25c1.PNG)
[octoprint.log](https://github.com/OctoPrint/OctoPrint/files/6241965/octoprint.log)
[octopi-1617264531106.log](https://github.com/OctoPrint/OctoPrint/files/6242030/octopi-1617264531106.log)
cant upload gcode file.

Terminal tab:
`Recv: ok
Send: N6392 G1 F3000 X108.865 Y68.314 E42.48993*58
Recv: ok
Send: N6393 G0 F9000 X107.252 Y68.313*68
Recv:   TT::210.43210.43  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::8383  BB@@::3030
Recv: 
Recv: ok
Send: N6394 G1 F3000 X84.913 Y151.684 E44.21237*50
Recv: ok
Send: N6395 G0 F9000 X85.841 Y151.684*65
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9191  BB@@::3030
Recv: 
Recv: ok
Send: N6396 G1 F3000 X58.691 Y124.534 E44.97861*58
Recv: ok
Send: N6397 G0 F9000 X58.692 Y125.239*66
Recv: ok
Send: N6398 G1 F3000 X161.306 Y97.744 E47.09862*63
Recv: ok
Send: N6399 G0 F9000 X161.307 Y98.401*68
Recv:   TT::209.92209.92  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9292  BB@@::3030
Recv: 
Recv: echo:busy: processing
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::104104  BB@@::3030
Recv: 
Recv: ok
Send: N6400 G1 F3000 X147.03 Y151.684 E48.19944*54
Recv: ok
Send: N6401 G0 F9000 X145.454 Y151.685*127
Recv: ok
Send: N6402 G1 F3000 X161.307 Y147.437 E48.52697*4
Recv: ok
Send: N6403 G0 F9000 X161.307 Y144.764*112
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::102102  BB@@::3030
Recv: 
Recv: ok
Send: N6404 G1 F3000 X159.453 Y151.684 E48.66994*9
Recv: ok
Send: N6405 G0 F9000 X161.307 Y142.297*121
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::102102  BB@@::3030
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6406 G1 F3000 X87.324 Y68.315 E50.75788*0
Recv: ok
Send: N6407 G0 F9000 X88.617 Y68.313*125
Recv: ok
Send: N6408 G1 F3000 X66.278 Y151.684 E52.48033*52
Recv: ok
Send: N6409 G0 F9000 X68.87 Y151.684*114
Recv:   TT::209.57209.57  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9898  BB@@::3030
Recv: 
Recv: ok
Send: N6410 G1 F3000 X58.691 Y141.505 E52.7676*4
Recv: ok
Send: N6411 G0 F9000 X58.692 Y143.874*72
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0160.01  //60.0060.00  @@::8989  BB@@::2626
Recv: 
Recv: ok
Send: N6412 G1 F3000 X161.306 Y116.379 E54.88761*15
Recv: ok
Send: N6413 G0 F9000 X161.306 Y110.167*116
Recv:   TT::210.04210.04  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9090  BB@@::3030
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6414 G1 F3000 X58.691 Y137.663 E57.00765*51
Recv: ok
Send: N6415 G0 F9000 X58.692 Y131.451*66
Recv: ok
Send: N6416 G1 F3000 X161.307 Y103.955 E59.12768*13
Recv: ok
Send: N6417 G0 F9000 X161.307 Y99.871*79
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7777  BB@@::3030
Recv: 
Recv: ok
Send: N6418 G1 F3000 X129.751 Y68.315 E60.01826*55
Recv: ok
Send: N6419 G0 F9000 X132.048 Y68.314*73
Recv: ok
Send: N6420 G1 F3000 X58.691 Y87.97 E61.53382*61
Recv: ok
Send: N6421 G0 F9000 X58.691 Y87.269*119
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7878  BB@@::3030
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6422 G1 F3000 X63.771 Y68.313 E61.92545*6
Recv: ok
Send: N6423 G0 F9000 X62.5 Y68.314*125
Recv:   TT::210.66210.66  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7777  BB@@::2929
Recv: 
Recv: ok
Send: N6424 G1 F3000 X58.691 Y69.335 E62.00415*14
Recv: ok
Send: N6425 G0 F9000 X61.869 Y68.315*123
Recv: ok
Send: N6426 G1 F3000 X145.238 Y151.684 E64.357*4
Recv: ok
Send: N6427 G0 F9000 X140.818 Y151.684*123
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7878  BB@@::2929
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6428 G1 F3000 X161.307 Y75.219 E65.93678*53
Recv: ok
Send: N6429 G0 F9000 X161.307 Y79.109*74
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9292  BB@@::2929
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6430 G1 F3000 X58.692 Y106.604 E68.0568*6
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9090  BB@@::2929
Recv: 
Recv: ok
Send: N6431 G0 F9000 X58.691 Y107.564*69
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::104104  BB@@::2929
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6432 G1 F3000 X102.811 Y151.684 E69.30197*3
Recv: ok
Send: N6433 G0 F9000 X103.548 Y151.684*113
Recv: ok
Send: N6434 G1 F3000 X125.887 Y68.313 E71.02441*57
Recv: ok
Send: N6435 G0 F9000 X121.265 Y68.315*73
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::102102  BB@@::2929
Recv: 
Recv: ok
Send: N6436 G1 F3000 X161.307 Y108.356 E72.15447*13
Recv: ok
Send: N6437 G0 F9000 X161.306 Y128.802*115
Recv: ok
Send: N6438 G1 F3000 X75.907 Y151.685 E73.91882*50
Recv: ok
Send: N6439 G0 F9000 X72.49 Y151.684*120
Recv: ok
Send: N6440 G1 F3000 X94.829 Y68.313 E75.64127*4
Recv: ok
Send: N6441 G0 F9000 X95.81 Y68.315*76
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::102102  BB@@::2929
Recv: 
Recv: echo:busy: processing
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0160.01  //60.0060.00  @@::103103  BB@@::2525
Recv: 
Recv: ok
Send: N6442 G1 F3000 X161.307 Y133.812 E77.48974*13
Recv: ok
Send: N6443 G0 F9000 X161.306 Y135.014*115
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::8888  BB@@::2929
Recv: 
Recv: ok
Send: N6444 G1 F3000 X99.09 Y151.685 E78.77513*10
Recv: ok
Send: N6445 G0 F9000 X97.336 Y151.684*67
Recv:   TT::210.04210.04  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9090  BB@@::2929
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6446 G1 F3000 X119.675 Y68.314 E80.49755*48
Recv: ok
Send: N6447 G0 F9000 X132.099 Y68.313*73
Recv: ok
Send: N6448 G1 F3000 X109.76 Y151.684 E82.22*5
Recv: ok
Send: N6449 G0 F9000 X122.272 Y151.685*112
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7777  BB@@::2929
Recv: 
Recv: ok
Send: N6450 G1 F3000 X161.307 Y141.225 E83.02647*15
Recv: ok
Send: N6451 G0 F300 X161.307 Y141.225 Z2.16*41
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7878  BB@@::2929
Recv: 
Recv: ok
Send: N6452 G0 F9000 X161.388 Y151.765*118
Recv: ok
Send: N6453 G1 F1500 X58.612 Y151.765 E85.07748*60
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7878  BB@@::2929
Recv: 
Recv: ok
Send: N6454 G1 X58.612 Y68.235 E86.74441*111
Recv: ok
Send: N6455 G1 X161.388 Y68.235 E88.79542*82
Recv:   TT::210.59210.59  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7979  BB@@::3030
Recv: 
Recv: ok
Send: N6456 G1 X161.388 Y151.765 E90.46235*104
Recv: ok
Send: N6457 G0 F9000 X161.788 Y152.165*114
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::9292  BB@@::3030
Recv: 
Recv: ok
Send: N6458 G1 F1500 X58.212 Y152.165 E92.52932*51
Recv: ok
Send: N6459 G1 X58.212 Y67.835 E94.21221*96
Recv:   TT::210.04210.04  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::8989  BB@@::3030
Recv: 
Recv: ok
Send: N6460 G1 X161.788 Y67.835 E96.27918*82
Recv: ok
Send: N6461 G1 X161.788 Y152.165 E97.96208*105
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::104104  BB@@::3030
Recv: 
Recv: ok
Send: N6462 G0 F9000 X162.188 Y152.565*117
Recv: ok
Send: N6463 G1 F1500 X57.812 Y152.565 E100.04502*12
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::102102  BB@@::3030
Recv: 
Recv: ok
Send: N6464 G1 X57.812 Y67.435 E101.74388*88
Recv: ok
Send: N6465 G1 X162.188 Y67.435 E103.82681*99
Recv: ok
Send: N6466 G1 X162.188 Y152.565 E105.52568*92
Recv: ok
Send: N6467 G0 F9000 X161.918 Y152.295*122
Recv: ok
Send: N6468 G0 X161.307 Y150.662*16
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::102102  BB@@::3030
Recv: 
Recv: echo:busy: processing
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::103103  BB@@::3131
Recv: 
Recv: echo:busy: processing
Recv:   TT::210.00210.00  //210.00210.00  BB::59.9959.99  //60.0060.00  @@::8989  BB@@::3535
Recv: 
Recv: ok
Send: N6469 G1 F3000 X78.959 Y68.315 E107.8497*13
Recv:   TT::210.08210.08  //210.00210.00  BB::60.0260.02  //60.0060.00  @@::8989  BB@@::2424
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6470 G0 F9000 X76.106 Y68.313*123
Recv:   TT::210.62210.62  //210.00210.00  BB::59.9959.99  //60.0060.00  @@::7777  BB@@::3535
Recv: 
Recv: echo:busy: processing
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::7979  BB@@::3131
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6471 G1 F3000 X58.692 Y133.303 E109.1924*52
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0260.02  //60.0060.00  @@::7878  BB@@::2323
Recv: 
Recv: echo:busy: processing
Recv:   TT::210.62210.62  //210.00210.00  BB::60.0260.02  //60.0060.00  @@::7878  BB@@::2222
Recv: 
Recv: ok
Send: N6472 G0 F9000 X58.691 Y132.9*74
Recv: ok
Send: N6473 G1 F3000 X77.476 Y151.684 E109.72254*5
Recv:   TT::210.55210.55  //210.00210.00  BB::60.0160.01  //60.0060.00  @@::7979  BB@@::2525
Recv: 
Recv: echo:busy: processing
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0160.01  //60.0060.00  @@::9090  BB@@::2424
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6474 G0 F9000 X78.613 Y151.684*66
Recv:   TT::209.61209.61  //210.00210.00  BB::59.9959.99  //60.0060.00  @@::9898  BB@@::3232
Recv: 
Recv: echo:busy: processing
Recv:   TT::209.37209.37  //210.00210.00  BB::59.9959.99  //60.0060.00  @@::102102  BB@@::3232
Recv: 
Recv: ok
Send: N6475 G1 F3000 X100.953 Y68.313 E111.44499*3
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0260.02  //60.0060.00  @@::101101  BB@@::2020
Recv: 
Recv: echo:busy: processing
Recv:   TT::209.37209.37  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::101101  BB@@::2828
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6476 G0 F9000 X104.415 Y68.315*72
Recv:   TT::209.45209.45  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::100100  BB@@::2828
Recv: 
Recv: echo:busy: processing
Recv:   TT::210.00210.00  //210.00210.00  BB::60.0060.00  //60.0060.00  @@::8888  BB@@::2828
Recv: 
Recv: ok
Send: N6477 G1 F3000 X161.307 Y125.206 E113.05059*63
Recv: ok
Send: N6478 G0 F9000 X161.307 Y122.678*112
Recv:   TT::210.47210.47  //210.00210.00  BB::59.9959.99  //60.0060.00  @@::7878  BB@@::3232
Recv: 
Recv: echo:busy: processing
Recv:   TT::210.62210.62  //210.00210.00  BB::59.9759.97  //60.0060.00  @@::7676  BB@@::3737
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6479 G1 F3000 X58.691 Y150.174 E115.17064*15
Recv:   TT::210.62210.62  //210.00210.00  BB::59.9759.97  //60.0060.00  @@::7777  BB@@::3838
Recv: 
Recv: echo:busy: processing
Recv: ok
Send: N6480 G0 F9000 X58.691 Y149.87*125
Recv:   TT::210.62210.62  //210.00210.00  BB::59.9759.97  //60.0060.00  @@::7777  BB@@::3939
Recv: 
`

**Comment by cp2004** [comment]
Can you confirm that this didn't happen in OctoPrint 1.5.3 or earlier?

It's well known that the analysis can be inaccurate, if it is showing a red dot beside it then OctoPrint *knows* it is inaccurate.

**Comment by foosel** [comment]
I didn't touch print time estimation between 1.5 and 1.6, so it cannot be a regression.

Sounds more like the file so non uniform that half way through the linear approximation diverges SO much from the estimate it will assume the estimate was wrong in some way and rather go with linear altogether. Would need the file however to confirm, so please share, at the very least in private to gina -at- octoprint.org

**Comment by foosel** [comment]
@mrspaceman still waiting for a file here to analyse. I can't look into this without that.

**Comment by mrspaceman** [comment]
Sorry, the print time issue happened with this (as well as all my other) files

[Everest.zip](https://github.com/OctoPrint/OctoPrint/files/6263183/Everest.zip)

**Comment by foosel** [comment]
Then please share some of the others as well. Looking at that it is no wonder that OctoPrint's attempt at linear estimating based on progress and calculated total will be deemed invalid after a while, the shape gets constantly smaller with a big AF bottom, so is anything but linear and thus a prime example of cases where the estimate will produce invalid results.

Have you tried the same files through 1.5? Is there a difference in plugins that are enabled and/or installed? Do you maybe used to have Print Time Genius installed and it didn't process these files?

Again, estimation wasn't really touched apart from one bug fix that I don't see how it could impact things here.

**Comment by mrspaceman** [comment]
heres another one 
[paint-dish_v2-1.zip](https://github.com/OctoPrint/OctoPrint/files/6266025/paint-dish_v2-1.zip)

**Comment by mrspaceman** [comment]
I am new to Octoprint and the only plugins I have installed are:
- FileManager
- Octolapse (disabled)
- Creality-2x-temperature-reporting-fix (0.0.4)
- Telegram Notifications

**Comment by foosel** [comment]
That still leaves open the question whether you see the same behaviour under 1.5.3 or not.

**Comment by mrspaceman** [comment]
I upgraded to 1.6.0rc1 as soon as I installed Octoprint, so I dont know if the same would happen in earlier version.

**Comment by foosel** [comment]
Then please downgrade to 1.5.0 and verify if the same issue exists there. 1.6.0rc1 - as the name suggests - is still a release candidate and not a stable release. I need to know whether the issue is something introduced by 1.6 (and thus a regression) or if it already existed in 1.5. The whole point of release candidates is to test new functionality and to identify regressions in existing functionality before stable rollout - so a comparison point with earlier stable releases is pretty important.

**Comment by foosel** [comment]
FWIW, I just took a look at the analysis results of the Everest file against 1.5.3 and 1.6.0rc2.dev.

1.5.3:

![firefox_2021-04-07_11-43-33](https://user-images.githubusercontent.com/83657/113846796-22861300-9797-11eb-8ee4-216bd6f459fd.png)

1.6.0rc2.dev:

![image](https://user-images.githubusercontent.com/83657/113846627-f8ccec00-9796-11eb-87a6-41c5be843820.png)

The only difference there is the calculated printed model height, and that is due to a bugfix in the model size calculation (#4036). 

With that, the shape of the model, the described issue and the screenshot this reeks of the usual issue with non uniform and non linear models where OctoPrint's analysis backed linear estimate while printing will be deemed invalid after it leaves the limits of sane expected values based on the estimate and fall back to a full linear estimate. 

Suggestion: Install the Print Time Genius plugin, that does some more (resource intensive) stuff than the bare minimum that OctoPrint can support for all possible runtime scenarios. If that *doesn't* fix it, request a reopen of this ticket. For now I'll close this as known issue and won't fix since the limitations of the print time estimation are well known, I've explained where the issues lie repeatedly here and elsewhere and there exists a solution that works for most people (the aforementioned plugin).
